src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -180,3 +181,12 @@ def _lock_cb(cls, mode, n, file, line):
 # condition registering the OpenSSL locks. On Python 3.4+ the import lock
 # is per module so this approach will not work.
 Binding.init_static_locks()
+
+if Binding.lib.SSLeay() < 0x10001000:
+    warnings.warn(
+        "OpenSSL versions less than 1.0.1 are no longer supported by the "
+        "OpenSSL project as of 2015-12-31. Your distribution/OS may provide "
+        "its own limited support, but we urge users to upgrade to a newer "
+        "OpenSSL as soon as possible.",
```

Comment:
I think we should go for a simpler message here, similar to what we do for Python 2.6, something that says, "Upstream has dropped support for this, it is no longer receiving security patches from them. Cryptography will drop support for this in an as-yet undetermined future release. Please upgrade" (but with cleaner language).

We should also add a CHANGELOG entry for increased visibility.